provider/pkg/provider/provider.go

```diff
@@ -1552,6 +1559,9 @@ func (k *kubeProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*p
 	newInputs := propMapToUnstructured(newResInputs)
 
 	oldInputs, oldLive := parseCheckpointObject(oldState)
+	if !isHelmRelease(urn) {
+		contract.Assertf(oldLive.GetName() != "", "expected live object name to be nonempty: %v", oldLive)
+	}
```

Comment:
This is a workaround for https://github.com/pulumi/pulumi-kubernetes/issues/2679 and I'll add a comment before merging.